### ISSUE TITLE
unfortunately myIonicApp has stopped

### ISSUE THREAD

**Issue by ranbuch** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#128]
Hi,

I'm using ionic 2 and cordova-plugin-camera-preview version 0.0.8

after running:
`ionic plugin add cordova-plugin-camera-preview`
`ionic build android`
`ionic emulate android`
`adb -s <deviceId> install <path>\android-debug.apk`

setting a picture taken handler:

``` javascript
constructor(private navCtrl: NavController) {
  CameraPreview.setOnPictureTakenHandler().subscribe((result) => {
    this.onCameraPreview(result);
  }
});

...
onCameraPreview(result: Array<any>) {
    this.originalPicture = result[0];
    this.previewPicture = result[0];
}
```

and setting a click event on some button:

``` javascript
startCapturing() {
    this.logs.push('startCapturing init');
    try {
      CameraPreview.startCamera({x: 50, y: 100, width: 300, height:300, camera: "back", tapPhoto: true, previewDrag: true, toBack: false, alpha: 1});
    }
    catch (e) {
      this.logs.push('CameraPreview.setOnPictureTakenHandler errors: ' + e);
    }
  }
```

The app is running both on my device (Samsung galaxy s7 edge) and Nexus 6P emulator.

When I'm clicking the button that is calling the `startCapturing()` function I'm getting:
`unfortunately myIonicApp has stopped`

Basically the app crashes . . .

Any idea what am I doing wrong here?

Thanks!

**Comment by ranbuch** [comment]
So terns out the app doesn't have the permissions to access the camera.

I think this is an android 6 issue as the developer should ask the user for permissions in real time.

I've granted the Camera permissions to this app manually for develop purposes (and it works!) but I believe you should update the plugin - witch is great by the why!

**Comment by diego-cortez** [comment]
Hi @ranbuch, how is that you solve the problem with android 6 permissions? I'm currently having this issue and i have no idea how to solve it, do you hace a fork or something where I can see your solution, Thank you very much in advance.

**Comment by ranbuch** [comment]
For development purposes only I've approved the camera permissions on my device settings after installing the APK.

After further development I've realized that there's a couple of issues with this component (on android at list) that I could not resolve:
Pixelated images (even when using the preview image diminutions as a reference for the preview size).
No focus control (or auto-focus callback).

That's what I can recall now anyway.

I just went ```Native``` :(

Best of luck!

**Comment by diego-cortez** [comment]
I solved the problem by using 'cordova-plugin-android-permissions' to force the app to ask user for permissions, works good. Anyway, what you said about pixelated images and focus control is totally true but I can't go native at this point, I will figurate out how to solve this kind of problems. Thank you very much for you answer I appreciate it.

**Comment by ranbuch** [comment]
Thank you for sharing your solution. Best of luck!

**Comment by westonganger** [comment]
Yes the permissions still need to be fixed for Android. I would love to see a PR but this is duplicate of a few other issues. Closing
